sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
 
   /**
    * Create a [[ShowTablesCommand]] logical plan.
+   * Example SQL :
+   * SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards']
    */
   override def visitShowTables(ctx: ShowTablesContext): LogicalPlan = withOrigin(ctx) {
-    if (ctx.LIKE != null) {
-      logWarning("SHOW TABLES LIKE option is ignored.")
+    if (ctx.qualifiedName() != null) {
+      logWarning("SHOW TABLES qualifiedName is ignored.")
```

Comment:
@hvanhovell @rxin 
qualifiedName does not seem to work in hive. Only the table name pattern works. Not sure if its a defect in hive...
Just for reference i looked at cloudera documentation on how they support SHOW commands. They just take
a pattern string literal. Should we simplify our grammer and drop the qualifiedName ? Please let me know.
http://www.cloudera.com/documentation/archive/impala/2-x/2-1-x/topics/impala_show.html